searcher/searcher.go

```diff
@@ -367,6 +365,13 @@ func newSearcher(
 		SpecialFiles:    wd.SpecialFiles(),
 	}
 
+	var vcsDir string
+	if (wd.IsLocal()) {
```

Comment:
I think this would better separate concerns if the vcs interface had a function kind of like this:

```
WorkDirFrom(dbpath string, repo *config.Repo) string
```